services/worker/src/worker/utils.py

```diff
@@ -284,6 +284,8 @@ def get_rows(
         raise TypeError("load_dataset should return a Dataset in normal mode")
     if column_names:
         ds = ds.select_columns(column_names)
+    if not rows_max_number:
+        return list(ds)
```

Comment:
You'll OOM if you do that, some datasets are hundreds of gigabytes.

You can use it as a generator instead of a list

(same for `rows_plus_one ` a few lines later - we can't use a list with the full dataset)